pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -124,7 +124,7 @@ public TableResizer(DataSchema dataSchema, QueryContext queryContext) {
    */
   private OrderByValueExtractor getOrderByValueExtractor(ExpressionContext expression) {
     if (expression.getType() == ExpressionContext.Type.LITERAL) {
-      return new LiteralExtractor(expression.getLiteral());
+      return new LiteralExtractor(expression.getLiteralString());
```

Comment:
i think we can change LiteralExtractor to actually extract the literal. 

all literals we supported now are comparable. thus we can carry the type into it

Reply:
Yeah. but I want to do this in next PR since there are too many refactoring stuff to do for this.